src/contexts/auth-provider.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable object-curly-spacing */
+import React, { useEffect, useState } from 'react';
+import { getCurrentUser, getUser } from '../firebase/api/users';
+
+const AuthContext = React.createContext();
+
+const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState();
+  useEffect(async () => {
+    try {
+      const auth = await getCurrentUser();
+      if (auth) {
+        const user = await getUser(auth.uid);
+        setUser(user.data);
+      }
+    } catch (err) {
```

Comment:
Should we alert to log in if there's an error?

Reply:
Yeah, we could like show that annoying message

"Username or password is incorrect"

Reply:
But that logic could probably go to the Login page, unless we encapsulate that functionality here and return an error with the provider. I think for now we leave it and create a ticket :)